src/zjs_sensor.c

```diff
@@ -0,0 +1,512 @@
+// Copyright (c) 2016, Intel Corporation.
+#ifdef BUILD_MODULE_SENSOR
+#ifndef QEMU_BUILD
+#ifndef ZJS_LINUX_BUILD
+// Zephyr includes
+#include <zephyr.h>
+#endif
+
+#include <string.h>
+
+// ZJS includes
+#include "zjs_sensor.h"
+#include "zjs_callbacks.h"
+#include "zjs_ipm.h"
+#include "zjs_util.h"
+
+#define ZJS_SENSOR_TIMEOUT_TICKS                      500
+
+static struct nano_sem sensor_sem;
+
+enum sensor_state {
```

Comment:
We discussed these changes back in the W3C TPAC meeting in Lisbon and they are slowly making it into the spec